src/pages/index.tsx

```diff
@@ -8,21 +10,54 @@ import task from '@/interfaces/task.type';
 import Accordion from '@/components/Accordion';
 
 const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
 
-function renderCardList(tasks: task[]) {
-  return tasks.map((item: task) => <Card content={item} key={item.id} />);
+async function updateCardContent(cardDetails: any) {
+  let response = {};
+  let error = {};
+  try {
+    response = await axios({
+      method: 'patch',
+      url: `https://api.realdevsquad.com/tasks/${cardDetails.id}`,
+      params: null,
+      data: cardDetails,
+      headers: {
+        'Content-type': 'application/json',
+      },
+      withCredentials: true,
+    });
+  } catch (err:any) {
+    error = err;
+    // eslint-disable-next-line no-console
+    console.log('Error', error);
+  }
+  // eslint-disable-next-line no-console
+  console.log('Response', response);
+}
+
+function renderCardList(tasks: task[], edit: boolean) {
+  return tasks.map((item: task) => (
+    <Card
+      content={item}
+      key={item.id}
+      shouldEdit={edit}
+      onContentChange={async (newDetails: any) => updateCardContent(newDetails)}
+    />
+  ));
 }
 
 const Index: FC = () => {
+  const router = useRouter();
+  const { query } = router;
   let tasks: task[] = [];
   const [filteredTask, setFilteredTask] = useState<any>([]);
-
   const { response, error, isLoading } = useFetch(TASKS_URL);
-
+  const [roles, setRoles] = useState(false);
+  const edit = !!query.edit && roles;
```

Comment:
Can you explain this?

Reply:
I used the state variable roles with initial value FALSE and then used it to initialize the edit(variable). That is, if query parameter is true and roles is also true, only then the user will be able to fill in and do edits

Reply:
Got it 👍🏽 
You need to name the variables with suitable names.
Think of a better name like `isUserAuthorized` or `isAdmin` or `isEditable`.